src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -1275,6 +1280,28 @@ impl OrchestratorWorker {
     }
 
     async fn run(mut self) {
+        {
+            info!("initializing Kubernetes orchestrator worker");
+            let start = Instant::now();
+
+            // Fetch the owner reference for our own pod (usually a
+            // StatefulSet), so that we can propagate it to the services we
+            // create.
+            let hostname = env::var("HOSTNAME").unwrap_or_else(|_| panic!("HOSTNAME environment variable missing or invalid; required for Kubernetes orchestrator"));
+            let orchestrator_pod = Retry::default()
+                .clamp_backoff(Duration::from_secs(10))
+                .retry_async(|_| self.pod_api.get(&hostname))
+                .await
+                .expect("always retries on error");
+            self.owner_references
+                .extend(orchestrator_pod.owner_references().into_iter().cloned());
```

Comment:
it's a bummer this isn't something we can easily inject into the pod :(

hopefully another small thing we can clean up if/when we pull out the orchestrator

Reply:
Yeah, I looked into using the downward API, but there's no way to propagate owner references. I also looked into just passing the owner reference to use via CLI args, but you end up with a circular reference; the StatefulSet can't refer to its own UID in the pod template spec. At least not in any way that I could find.